components/src/CartCreate/cart.create.tsx

```diff
@@ -0,0 +1,357 @@
+              <label htmlFor="cart_edit">Name</label>
+              <input type="text" value={cartName} id="cart_edit" className="cart-edit-field" onChange={this.handleChange} />
+              {cartName.length > 0 && (<span role="presentation" className="clear-field-btn" onClick={this.clearCartNameField} />)}
+            </div>
+            <div className="btn-container">
+              <button type="button" className="ep-btn cancel-btn" onClick={this.handleHideCartForm}>{intl.get('cancel')}</button>
+              <button type="button" className="ep-btn primary save-btn" onClick={this.handleAddNewCart}>{intl.get('save')}</button>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  handleShowCartForm() {
+    this.setState({ showAddNewCartForm: true });
+  }
+
+  handleHideCartForm() {
+    this.setState({ showAddNewCartForm: false });
+  }
+
+  handleEditCart(index) {
+    const { cartElements } = this.state;
+    const elements = [...cartElements];
+    elements[index] = { ...elements[index] };
+    elements[index].editMode = true;
+    this.setState({ cartElements: elements });
+  }
+
+  handleCancelEditCart(index) {
+    const { cartElements } = this.state;
+    const elements = [...cartElements];
+    elements[index] = { ...elements[index] };
+    elements[index].editMode = false;
+    this.setState({ cartElements: elements });
+  }
+
+  createNewCart() {
+    const { cartName } = this.state;
+    const { handleCartsUpdate } = this.props;
+    this.setState({ showLoader: true });
+    login().then(() => {
+      cortexFetch(`/carts/${Config.cortexApi.scope}/form?followlocation&format=standardlinks,zoom.nodatalinks`, {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+        },
+        body: JSON.stringify({ name: cartName }),
+      })
+        .then(res => res.json())
+        .then((res) => {
+          this.fetchCartData();
+          handleCartsUpdate();
+          this.setState({
+            cartName: '',
+            showAddNewCartForm: false,
+            showLoader: false,
+          });
+        })
+        .catch((error) => {
+          // eslint-disable-next-line no-console
+          console.error(error.message);
+        });
+    });
+  }
+
+  render() {
+    const { cartElements, showAddNewCartForm } = this.state;
+    const {
+      handleModalClose, openModal,
+    } = this.props;
+
+    return (
+      <Modal open={openModal} onClose={handleModalClose}>
+        <div className="modal-lg cart-create-modal">
+          <div className="modal-content">
+            <div className="modal-header">
+              <h2 className="modal-title">
+                {intl.get('manage-carts')}
+              </h2>
+            </div>
+            <div className="modal-body">
+              <div className="create-cart-btn-wrap">
```

Comment:
We should only show the option of creating a new cart if the `createcartform` is returned in the response.